lib/3scale/backend/stats/cleaner.rb

```diff
@@ -197,6 +221,30 @@ def service_from_stats_key(stats_key)
             # simply ignore those keys.
             nil
           end
+
+          def delete_stats_keys_with_val_0(redis_conn, log_deleted_keys)
+            cursor = 0
+
+            loop do
+              cursor, keys = redis_conn.scan(cursor, count: SCAN_SLICE)
+
+              stats_keys = keys.select { |k| is_stats_key?(k) }
+
+              unless stats_keys.empty?
+                values = redis_conn.mget(*(stats_keys.to_a))
+                to_delete = Hash[stats_keys.zip(values)].select { |_k, v| v == '0'.freeze }
```

Comment:
You don't really need to create a `Hash` (or 2) here. How about `stats_keys.zip(values).select { |_, v| v == '0'.freeze }.map(&:first)`? You could even do these in place in the zipped array.

Reply:
You're right, I forgot about this. The values are needed in the other method because it prints keys and values, but there's no need to print the values in this one because they're always 0.

Reply:
Fixed :+1: